### PR TITLE
Feat : Dashboard screen

### PR DESCRIPTION
**Pull request by Ruthvikbr**
# PR Checklist

## Add description below this heading

- PR contains dashboard screen changes such as  Offers carousel , Popular menu items , coffee news and starbucks news

## Info

- [x] This PR can be merged
- [ ] Added new tests to cover new/updated code
- [x] I have attached screenshots/recordings for UI related PRs
- [ ] Documentation update is required
- [ ] Documentation has been updated

## PR Type

- [ ] Bug Fix
- [x] New Feature
- [ ] Enhancement

https://user-images.githubusercontent.com/37804253/196953336-25e709e3-bc03-4bbe-9ead-7f3163924108.mp4

### CODE REVIEW

File: app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/screens/home/HomeViewModel.kt [hunk 1]
```diff
@@ -41,4 +48,10 @@ class HomeViewModel @Inject constructor(
             _popularMenuItemsList.value = fetchPopularMenuItemsUseCase()
         }
     }
+
+    private fun fetchStarbucksNewsItems() {
+        viewModelScope.launch {
```
Comment (oianmol) [inline app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/screens/home/HomeViewModel.kt:53]: @Ruthvikbr how are we doing error handling here ?

File: app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/components/StarbucksColumn.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun StarbucksColumn(children: @Composable () -> Unit) {
```
Comment (pushpalroy) [inline app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/components/StarbucksColumn.kt:15]: @Ruthvikbr Should the children use a ColumnScope here?

Comment (Ruthvikbr) [reply app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/components/StarbucksColumn.kt:15]: @pushpalroy The idea here was to implement a reusable column component. Hence I tried this

File: app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/screens/home/HomeViewModel.kt (hunk 1 above)
Comment (Ruthvikbr) [reply app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/screens/home/HomeViewModel.kt]: @oianmol I have made use of CoroutineExceptionHandler as suggested

File: app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/components/BottomNavigationBar.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        items.forEach { item ->
+            val isSelected = currentDestination?.hierarchy?.any { it.route == item.route } == true
+            BottomNavigationItem(
+                icon = {
+                    Icon(
+                        painter =  painterResource(
+                            id = item.icon
+                        ),
+                        contentDescription = item.label,
+                        modifier = Modifier.size(30.dp),
```
[the remark this answers is not in the document]
Comment (Ruthvikbr) [reply app/src/main/java/com/ruthvikbr/starbucksindiacompose/ui/components/BottomNavigationBar.kt]: @pushpalroy I have updated this.

**Pull request merged**